cpp/src/arrow/flight/sql/server.cc

```diff
@@ -1072,6 +1260,18 @@ Status FlightSqlServerBase::EndTransaction(const ServerCallContext& context,
   return Status::NotImplemented("EndTransaction not implemented");
 }
 
+arrow::Result<ActionSetSessionOptionsResult> FlightSqlServerBase::SetSessionOptions (
+    const ServerCallContext& context,
+    const ActionSetSessionOptionsRequest& request) {
+  return Status::NotImplemented("SetSessionOptions not implemented");
+}
```

Comment:
Put another way: how do I actually _implement_ this? I don't get anything identifying the session here.

Reply:
* Handler would request access to the session object from server middleware.
* Middleware would either:
  * Already have found a valid session id cookie on call entry and set the session object for that call-local middleware instance
  * Or, if the session id cookie was not provided by the client the server session middleware would generate a new session object and corresponding ID
* Middleware would then set the response Set-Cookie header accordingly, and return the new OR looked-up session object to the handler.